test/WebJobs.Script.Tests.Integration/SamplesEndToEndTests.cs

```diff
@@ -290,6 +290,83 @@ public SamplesEndToEndTests(TestFixture fixture)
             Assert.Equal("Testing", TestHelpers.RemoveByteOrderMarkAndWhitespace(result));
         }
 
+        [Fact]
+        public async Task HttpTrigger_CSharp_UserAuth()
+        {
+            string uri = $"api/httptrigger-csharp-userauth?code={MasterKey}";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, uri);
+
+            MockEasyAuth(request, "facebook", "Connor McMahon", "10241897674253170");
+
+            HttpResponseMessage response = await this._fixture.HttpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            string responseContent = await response.Content.ReadAsStringAsync();
+            string[] identityStrings = StripBookendQuotations(responseContent).Split(';');
+            Assert.Equal("Identity: (facebook, Connor McMahon, 10241897674253170)", identityStrings[0]);
+            Assert.Equal("Identity: (key, Admin, master)", identityStrings[1]);
+
+            // verify that a request fails if it is totally unauthenticated
+            request = new HttpRequestMessage(HttpMethod.Get, "api/httptrigger-csharp-userauth");
+            response = await this._fixture.HttpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+
+            // verify that a request also fails using a function key
+            // when no user identity is provided
+            request = new HttpRequestMessage(HttpMethod.Get, $"api/httptrigger-csharp-userauth?code={MasterKey}");
+            response = await this._fixture.HttpClient.SendAsync(request);
+            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task HttpTrigger_UserAuth()
```

Comment:
Is this test a subset of the previous test?

Reply:
This tests the Node version of the function (HttpTrigger-UserAuth). Since the previous test covers the authentication that the "user" authLevel mode provides, this test was just to make sure that the new Node bindings work.

---
In reply to: [176888064](https://github.com/Azure/azure-functions-host/pull/2577#discussion_r176888064) [](ancestors = 176888064)